### PR TITLE
ASG invalid filter - whitespace problems

### PR DESCRIPTION
**Pull request by scotwk**
#947 

@kapilt 

I was having trouble replicating what the issue was asking for because all my ASGs are being reported as invalid locally, but I think this fix is what the issue was asking for.

But how diligent do we want to be when enforcing whitespace checks?  This exact same issue can probably occur on other fields (such as `ImageId` or `SecurityGroups`) causing an ASG to appear invalid.  Do we need to add checks for all the fields we access from the response?

### CODE REVIEW

File: c7n/resources/asg.py [hunk 1]
```diff
@@ -286,7 +286,10 @@ def get_asg_errors(self, asg):
         for bd in cfg['BlockDeviceMappings']:
             if 'Ebs' not in bd or 'SnapshotId' not in bd['Ebs']:
                 continue
-            if bd['Ebs']['SnapshotId'] not in self.snapshots:
+            snapshot_id = bd['Ebs']['SnapshotId']
+            if hasattr(snapshot_id, 'strip'):
```
Comment (kapilt) [inline c7n/resources/asg.py]: hasattr is quite implicit and heavy. is this just defensive? because the lines above it already handle the conditional.

Comment (scotwk) [inline c7n/resources/asg.py]: yeah - was just being defensive.  I can remove that if you feel confident it will always be a string.

Comment (kapilt) [inline c7n/resources/asg.py]: i'm confident, its a not-present or string value, in general on defensive of this sort, its better due to isinstance to be more explicit.

**Comment by scotwk** [comment]
@kapilt - sorry for the delay - I was traveling.  I `strip`ed all the fields we are checking in the invalid filter.

**Comment by kapilt** [comment]
no worries, and thanks looks good.

**Pull request merged**